File: fast_mm_matlab/src/utils/Rot3c.m
% Same as Rot3, but for cell-array.

function x = Rot3c(x, n)
  if nargin < 2
    n = 1;
  end
  n = mod(n, 3);
  switch n
  case 0
    return;
  case 1
    x = {x{2}, x{3}, x{1}};
    return;
  case 2
    x = {x{3}, x{1}, x{2}};
    return;
  end
end

File: fast_mm_python/src/utils/Rot3c.py
"""
Same as Rot3, but for list/tuple (cell-array equivalent).
"""


def Rot3c(x, n=1):
    """
    Same as Rot3, but for list/tuple (cell-array equivalent).
    
    Args:
        x: 3-element list or tuple
        n: Number of rotations (default: 1)
    
    Returns:
        Rotated list
    """
    n = n % 3
    
    if n == 0:
        return x
    elif n == 1:
        return [x[1], x[2], x[0]]
    elif n == 2:
        return [x[2], x[0], x[1]]

